src/odemis/driver/lakeshore.py

```diff
@@ -344,98 +411,220 @@ def HeaterSetup(self,
             2 = Medium, 3 = High
             For Output 2 in Voltage mode: 0 = Off, 1 = On
         """
-        self._sendOrder(b"RANGE %d,%d" % (self._output_channel, hrange))
+        self._sendOrder(b"RANGE %d,%d" % (output_channel, hrange))
 
-    def GetHeaterRange(self):
+    def GetHeaterRange(self, output_channel):
         """
         Query heater range enum int from the device
         reutrns: (int): Depending on setup, for Outputs 1 and 2 in Current mode: 0 = Off, 1 = Low, 
             2 = Medium, 3 = High
             For Output 2 in Voltage mode: 0 = Off, 1 = On
         """
-        return int(self._sendQuery(b'RANGE? %d' % (self._output_channel,)))
+        return int(self._sendQuery(b'RANGE? %d' % (output_channel,)))
 
     # Internal API functions
-    def _set_targetTemperature(self, value):
+    def _set_targetTemperature(self, value, output_channel):
         """
-        Setter for the targetTemperature VA
+        Setter for the targetTemperature VA.
         VA is in Celsius, but controller uses Kelvin
         """
         value = float(value)
-        self.SetSetpoint(value)
+        self.SetSetpoint(output_channel, value)
         # Read back the new value from the device
-        svalue = self.GetSetpoint()
+        svalue = self.GetSetpoint(output_channel)
         self.checkError()
         if abs(value - svalue) >= 0.01:
             logging.warning("Did not set new target temperature to %f", value)
         return svalue
 
-    def _set_heating(self, value):
+    def _set_heating(self, value, output_channel):
         """
         Setter for the heater range VA
         """
         value = int(value)
-        self.SetHeaterRange(value)
+        self.SetHeaterRange(output_channel, value)
         # Read back the new value from the device
-        svalue = self.GetHeaterRange()
+        svalue = self.GetHeaterRange(output_channel)
         self.checkError()
         if svalue != value:
             logging.warning("Did not set new heating range to %d", value)
         return svalue
 
     def _poll(self):
-        '''
+        """
         This method runs in a separate thread and polls the device for the temperature
-        '''
+        """
         try:
-            temp = self.GetSensorTemperature()
-            self.temperature._set_value(temp, force_write=True)
-            logging.debug(u"Lakeshore temperature: %f °C", self.temperature.value)
-            self.checkError()
+            for va_name, sensor_name in self._sensor_inputs.items():
+                temp = self.GetSensorTemperature(sensor_name)
+                if va_name == "":
+                    full_va_name = "temperature"
+                else:
+                    full_va_name = f"{va_name}Temperature"
+                va = getattr(self, full_va_name)
+                va._set_value(temp, force_write=True)
+                logging.debug(u"Lakeshore temperature: %f °C", va.value)
```

Comment:
You should extend the message with the sensor (or va) name. Otherwise we're going to get multiple identical log messages for different sensors.